packages/skill-api/src/core/services/stripe-checkout.ts

```diff
@@ -32,33 +33,46 @@ export async function stripeCheckout({
 
     try {
       const {getMerchantCoupon} = getSdk()
+
+      const querySchema = z
+        .object({
+          productId: z.string(),
+          upgradeFromPurchaseId: z.string().optional(),
+          userId: z.string().optional(),
+          couponId: z.string().optional(),
+          quantity: z.number().optional(),
+          bulk: z.boolean().default(false),
+        })
+        .transform(({quantity, userId, ...rest}) => {
+          return {
+            userId: userId || token?.sub,
+            quantity: quantity || 1,
+            ...rest,
+          }
+        })
```

Comment:
Added `Zod` parsing of the incoming query params which also sets defaults in the `transform`. This means we have typed values throughout this function instead of `any`s, so we no longer have to stick ` as string` all over the place.